libraries/Microsoft.Bot.Builder/BotState.cs

```diff
@@ -294,9 +261,10 @@ public BotStatePropertyAccessor(BotState botState, string name)
             /// <param name="turnContext">The turn context.</param>
             /// <param name="cancellationToken">The cancellation token.</param>
             /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
-            public Task DeleteAsync(ITurnContext turnContext, CancellationToken cancellationToken)
+            public async Task DeleteAsync(ITurnContext turnContext, CancellationToken cancellationToken)
             {
-                return _botState.DeletePropertyValueAsync(turnContext, Name, cancellationToken);
+                await _botState.LoadAsync(turnContext, false, cancellationToken).ConfigureAwait(false);
+                await _botState.DeletePropertyValueAsync(turnContext, Name, cancellationToken).ConfigureAwait(false);
```

Comment:
Awesome find! Please add a comment as to why this is happening, so the next person to look at the code doesn't need an in-person explanation...

Reply:
It might prudent to document the whole lazy load and save aspects of the the BotState BotStatePropertyAccessor.